backend/src/services/graph_access/graph_access.py

```diff
@@ -20,18 +20,26 @@ async def _request(self, url: str) -> httpx.Response:
             )
             return response
 
-    async def get_user_profile_photo(self) -> str | None:
-        print("entering get_user_profile_photo")
-        response = await self._request("https://graph.microsoft.com/v1.0/me/photo/$value")
+    async def get_user_profile_photo(self, user_id: str) -> str | None:
+        request_url = f"https://graph.microsoft.com/v1.0/me/photo/$value"
+
+        if user_id != "me":
+            request_url = f"https://graph.microsoft.com/v1.0/users/{user_id}/photo/$value"
+
+        response = await self._request(request_url)
 
         if response.status_code == 200:
             return base64.b64encode(response.content).decode("utf-8")
         else:
             return None
 
-    async def get_user_info(self) -> Mapping[str, str] | None:
-        print("entering get_user_info")
-        response = await self._request("https://graph.microsoft.com/v1.0/me")
+    async def get_user_info(self, user_id: str) -> Mapping[str, str] | None:
+        request_url = f"https://graph.microsoft.com/v1.0/me"
+
+        if user_id != "me":
+            request_url = f"https://graph.microsoft.com/v1.0/users/{user_id}"
```

Comment:
Same her. Consider if they should be treated "equally", or if one of them should be "default" and overridden if not "me".